p2p/src/test/kotlin/maru/p2p/P2PTest.kt

```diff
@@ -432,6 +462,77 @@ class P2PTest {
     }
   }
 
+  @Test
+  fun `peer can send beacon blocks by range request`() {
+    // Set up beacon chain with some blocks
+    val testBeaconChain = InMemoryBeaconChain(DataGenerators.randomBeaconState(number = 0u, timestamp = 0u))
+    val storedBlocks =
+      (0UL..10UL).map { blockNumber ->
+        DataGenerators.randomSealedBeaconBlock(number = blockNumber)
+      }
+
+    testBeaconChain.newUpdater().use { updater ->
+      storedBlocks.forEach { block ->
+        updater.putSealedBeaconBlock(block)
+      }
+      updater.putBeaconState(
+        BeaconState(
+          latestBeaconBlockHeader = storedBlocks.last().beaconBlock.beaconBlockHeader,
+          validators = DataGenerators.randomValidators(),
+        ),
+      )
+      updater.commit()
+    }
+
+    val p2PNetworkImpl1 =
+      P2PNetworkImpl(
+        privateKeyBytes = key1,
+        p2pConfig = P2P(ipAddress = IPV4, port = PORT1, staticPeers = emptyList()),
+        chainId = chainId,
+        serDe = RLPSerializers.SealedBeaconBlockSerializer,
+        metricsFacade = TestMetrics.TestMetricsFacade,
+        statusMessageFactory = statusMessageFactory,
+        beaconChain = testBeaconChain,
+        nextExpectedBeaconBlockNumber = initialExpectedBeaconBlockNumber,
+      )
+    val p2pManagerImpl2 =
+      P2PNetworkImpl(
+        privateKeyBytes = key2,
+        p2pConfig = P2P(ipAddress = IPV4, port = PORT2, staticPeers = listOf(PEER_ADDRESS_NODE_1)),
+        chainId = chainId,
+        serDe = RLPSerializers.SealedBeaconBlockSerializer,
+        metricsFacade = TestMetrics.TestMetricsFacade,
+        statusMessageFactory = statusMessageFactory,
+        beaconChain = beaconChain,
+        nextExpectedBeaconBlockNumber = initialExpectedBeaconBlockNumber,
+      )
+    try {
+      p2PNetworkImpl1.start()
+      p2pManagerImpl2.start()
+
+      awaitUntilAsserted { assertNetworkHasPeers(network = p2PNetworkImpl1, peers = 1) }
+      awaitUntilAsserted { assertNetworkHasPeers(network = p2pManagerImpl2, peers = 1) }
+
+      val peer1 =
+        p2pManagerImpl2.peerLookup.getPeer(LibP2PNodeId(PeerId.fromBase58(PEER_ID_NODE_1)))
+          ?: throw IllegalStateException("Peer with ID $PEER_ID_NODE_1 not found in p2pManagerImpl2")
+      val maruPeer1 = DefaultMaruPeer(peer1, rpcMethods, statusMessageFactory)
```

Comment:
peer1 is already is a `DefaultMaruPeer` and I think this means that user of p2pManagerImpl2 instance doesn't need the `rpcMethods`. Same applies to the other usage of `rpcMethods`, meaning that we should be able to remove it